src/Nethermind/Nethermind.Init/Steps/LoadGenesisBlock.cs

```diff
@@ -69,22 +71,20 @@ protected virtual void Load()
 
             ManualResetEventSlim genesisProcessedEvent = new(false);
 
-            bool genesisLoaded = false;
             void GenesisProcessed(object? sender, BlockEventArgs args)
             {
                 if (_api.BlockTree is null) throw new StepDependencyException(nameof(_api.BlockTree));
                 _api.BlockTree.NewHeadBlock -= GenesisProcessed;
-                genesisLoaded = true;
                 genesisProcessedEvent.Set();
             }
 
             _api.BlockTree.NewHeadBlock += GenesisProcessed;
             _api.BlockTree.SuggestBlock(genesis);
-            genesisProcessedEvent.Wait(TimeSpan.FromSeconds(40));
+            bool genesisLoaded = genesisProcessedEvent.Wait(_genesisProcessedTimeout);
 
             if (!genesisLoaded)
             {
-                throw new BlockchainException("Genesis block processing failure");
+                throw new TimeoutException($"Genesis block was not processed after {_genesisProcessedTimeout.TotalSeconds} seconds");
```

Comment:
Maybe we should terminate the app?

Reply:
We already do that. With this change the error message and exception are more descriptive about the issue